cmd/miniooni/main.go

```diff
@@ -145,7 +143,20 @@ type logHandler struct {
 }
 
 func (h *logHandler) HandleLog(e *log.Entry) (err error) {
-	s := fmt.Sprintf("<%s> %s", e.Level, e.Message)
+	if !globalOptions.verbose && e.Level != log.InfoLevel &&  e.Level != log.FatalLevel {
+		if e.Level > log.InfoLevel && !wrong {
+			log.Info(" O_o  Oh! Something is wrong." +
+				" Please try again with the -v option.")
+			wrong = true
+		}
+		return nil
+	}
+	if e.Level > log.InfoLevel && !wrong{
+		wrong = true
+	}
```

Comment:
Deciding whether there was an error based on the level of logging we see strikes me as weird. I believe the default action on `FatalLevel` is to `exit(1)` anyway, so there is already a clear indication from the Unix shell that something went wrong (and also PowerShell has that). For this reason, I would recommend to remove the part of the implementation dealing with the `wrong` variable.

Reply:
This part helps prevent repetition of the message in `non-verbose` mode:
https://github.com/ooni/probe-engine/blob/c8e30597c66f5d05cb79486450f3ecdf18c8adb0/cmd/miniooni/main.go#L147-L151

This part helps to print the message of success or failure at the end of the program in `verbose` mode:
https://github.com/ooni/probe-engine/blob/c8e30597c66f5d05cb79486450f3ecdf18c8adb0/cmd/miniooni/main.go#L154-L156

Remove?

However, this part `e.Level > log.InfoLevel &&` of this line should be deleted:
https://github.com/ooni/probe-engine/blob/c8e30597c66f5d05cb79486450f3ecdf18c8adb0/cmd/miniooni/main.go#L147-L148

Reply:
So, if the logger is verbose, you see verbose messages. If it's not, then you don't see them. This is the Unix convention that a program behaves as you wish and does not try to guess what you don't know (e.g. that there are verbose messages you can access if you specify `-v`).

With respect to `wrong`: you cannot decide whether there was a real error based on the log level you see. You can however count the number of failed experiments and used that information.